### PR TITLE
 Single File Installer

### LINKED ISSUE #93: Single file installer

### PR DESCRIPTION
**Pull request by mikemhenry**
Taken from https://github.com/openforcefield/openff-toolkit/blob/master/.github/workflows/installer.yml
h/t @j-wags

We will need to wait until the package is on conda-forge to test this.
Resolves #93

### CODE REVIEW

File: .github/workflows/installer.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      matrix:
+        os: [macOS-latest, ubuntu-latest]
+        python-version: [3.8, 3.9]
+
+    env:
+      CI_OS: ${{ matrix.os }}
+      PYVER: ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (IAlibay) [inline .github/workflows/installer.yaml:23]: 😱 is v3 out, I need to update many things.

**Review by IAlibay — COMMENTED** [review]
Overall this seems fine along maybe a little bit of cleaning up on the final step would be good. I'll hold off on approving since it's not immediately clear to me that this is the final version - @mikemhenry is this meant to be good to go?

**Comment by mikemhenry** [comment]
> Overall this seems fine along maybe a little bit of cleaning up on the final step would be good. I'll hold off on approving since it's not immediately clear to me that this is the final version - @mikemhenry is this meant to be good to go?

Sorry I need to add [WIP} to this!

**Comment by mikemhenry** [comment]
I've got this working on my fork, it was easier since github won't let me run triggered workflows from a PR. A few notes:

* Can I use the cudatoolkit shim so make the download smaller? Or will people want to run this on a GPU?
* I injected some extra dependencies for the notebook to work, but `cyjupyter` isn't on conda-forge so it isn't easy to get that added, but I think we can remove it from the notebook.

File: .github/workflows/installer.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        export LATEST_TAG=$(git ls-remote --tags https://github.com/openfreeEnergy/openfe.git | cut -f2 | grep -v "rc" | tail -1 | sed 's/refs\/tags\///')
+        echo $LATEST_TAG
+        echo $FOUND_VER
+        if [[ $LATEST_TAG != $FOUND_VER ]];
+          then echo "Version mismatch"
+          exit 1
+        fi
+        # This should work if run directly at release time, but a safer approach would be
+        # to `git checkout` the corresponding tag of the release, see PR #577
+        pytest -v ../openfe/tests/
```
[the remark this answers is not in the document]
Comment (mikemhenry) [reply .github/workflows/installer.yaml:97]: It wouldn't hurt but I'd like to keep this as close to "upstream" (openff) as possable.

**Comment by mikemhenry** [comment]
This is ready for review!

**Comment by mikemhenry** [comment]
This repo is also used: https://github.com/OpenFreeEnergy/toolkit-installer-constructor
I just changed it to work with openfe instead of openff + added some extra deps so the notebook works

**Comment by mikemhenry** [comment]
If there are no objections, I will merge this in EOD tomorrow (3/31)

**Comment by richardjgowers** [comment]
Thanks @mikemhenry !

**Pull request merged**